src/components/SingleArticle.jsx

```diff
@@ -41,17 +43,19 @@ function SingleArticle() {
         alt="Article"
       />
       <p className="article-body">{article.body}</p>
+      <p>{article.votes} votes</p>
 
-      <button onClick={handleViewComments}>
-        {comments.length === 0 ? "View Comments" : "Hide Comments"}
+      <button onClick={() => handleViewComments()}>
+        {showComments ? "Hide Comments" : "View Comments"}
       </button>
-      {comments.map((comment) => (
-        <div key={comment.id} className="comment-card">
-          <p>{comment.author}</p>
-          <p>posted on {comment.created_at}</p>
-          <p>{comment.body}</p>
-        </div>
-      ))}
+      {showComments &&
+        comments.map((comment, index) => (
```

Comment:
Individual comments should be rendered in a separate component, like a 'CommentCard'.